3.x: Add `usesTablets` to `KeyspaceMetadata`

Bouncheck opened the pull request:
Adds the field to `KeyspaceMetadata` that informs if the keyspace has tablets
enabled. It is initialized with value `false` and updated to `true` only if
the keyspace is discovered in `system_schema.scylla_keyspaces` and its entry
has non-null value in `initial_tablets` column. The update happens on schema
refresh.

Updates `Metadata#getReplicas` to use this keyspace metadata information to
decide whether to look for replicas in the token map or tablet map.

Fixes https://github.com/scylladb/java-driver/issues/380.

dkropachev commented on driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java:
```diff
@@ -1589,6 +1643,21 @@ SystemRows fetchSystemRows(
       }
 
+      if (isSchemaOrKeyspace) {
+        if (targetType == KEYSPACE) {
+          scyllaKsFuture =
+              queryAsync(
+                  SELECT_SCYLLA_KEYSPACES
+                      + " WHERE keyspace_name = '"
+                      + targetKeyspace
+                      + "' LIMIT 1;",
+                  connection,
+                  protocolVersion);
+        } else {
+          scyllaKsFuture = queryAsync(SELECT_SCYLLA_KEYSPACES, connection, protocolVersion);
+        }
```
`scylla_keyspaces` is not populated with keyspaces created with default configuration, so it is going to fail.

Bouncheck replied:
In that case it's going to return 0 rows, which is not breaking unless other code explicitly calls for a row that does not exist. 
I'm not sure if I understand the meaning of default here. Scylla 6.2.0 has tablets enabled by default and keyspaces created without explicit CQL syntax to enable tablets do appear in `scylla_keyspaces`. The keyspaces that do not appear are treated as not tablets-based.

Bouncheck committed:
Add `usesTablets` property to `KeyspaceMetadata`

Adds the field to `KeyspaceMetadata` that informs if the keyspace has tablets
enabled. It is initialized with value `false` and updated to `true` only if
the keyspace is discovered in `system_schema.scylla_keyspaces` and its entry
has non-null value in `initial_tablets` column. The update happens on schema
refresh.

Updates `Metadata#getReplicas` to use this keyspace metadata information to
decide whether to look for replicas in the token map or tablet map.

Fixes #380.

dkropachev commented on driver-core/src/main/java/com/datastax/driver/core/Metadata.java:
```diff
@@ -558,26 +558,31 @@ public Set<TokenRange> getTokenRanges(String keyspace, Host host) {
+    if (partitioner == null && current != null) {
+      partitioner = current.factory;
+    }
+    if (partitioner != null) {
```
Please return empty set if partitioner is null

dkropachev commented on driver-core/src/main/java/com/datastax/driver/core/Metadata.java:
```diff
@@ -558,26 +558,32 @@ public Set<TokenRange> getTokenRanges(String keyspace, Host host) {
+    table = handleId(table);
     TokenMap current = tokenMap;
-    if (current == null) {
+    Token token = null;
```
Please remove it.
